### PR TITLE
Javacli79

### PR DESCRIPTION
**Pull request by berkeleysquare**
added --log-verbose, --log-trace, and --log-debug which  mirror --verbose, -trace, and --debug but write to ./ds3_logs/spectra.log. Using the ch.qos.logback.core.rolling.TimeBasedRollingPolicy previous days' logs will be archived to spectraYYYY-MM-DD.log.

### CODE REVIEW

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/Arguments.java [hunk 1]
```diff
@@ -72,8 +69,17 @@
     private boolean followSymlinks = false;
     private ImmutableMap<String, String> metadata = null;
     private ImmutableMap<String, String> modifyParams = null;
+    private Level consoleLogLevel;
+    private Level fileLogLevel;
+
+    // don't use Logger because teh user's [refebrnces are not yet set
```
Comment (RachelTucker) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/Arguments.java]: misspellings: *the and *references

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/Arguments.java [hunk 2]
```diff
@@ -72,8 +69,17 @@
...
+    // collect log info that will be logged by Main
+    private static StringBuilder argumentLog = new StringBuilder("Argument processing");
+    private void addToLog(String logItem) { argumentLog.append(" | " + logItem) ; }
```
Comment (RachelTucker) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/Arguments.java]: missing final

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/Arguments.java (hunk 1 above)
Comment (berkeleysquare) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/Arguments.java]: Even worse, I was shooting for "preferences." Sigh.

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/Main.java [hunk 3]
```diff
@@ -23,17 +26,38 @@
...
-
-    private final static Logger LOG = LoggerFactory.getLogger(Main.class);
+    private final static ch.qos.logback.classic.Logger LOG = (ch.qos.logback.classic.Logger) LoggerFactory.getLogger(Main.class);
 
     public static void main(final String[] args) {
+
         try {
             final Arguments arguments = new Arguments(args);
 
+            // turn root log wide open, filters will be set to argument levels
```
Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/Main.java]: Please move the log configuration into a separate method.

**Pull request merged**